docs/sql_reference/src/asciidoc/_chapters/sql_functions_and_expressions.adoc

```diff
@@ -2325,82 +2328,206 @@ following abbreviations:
+==== The first day of week
+
+* This value cannot be specified, the default value is Sunday.
+
+* The DATEDIFF function returns the number of Sundays between _startdate_ and _enddate_. This function counts _enddate_ if it falls on Sunday, but doesn't count _startdate_ even if it does fall on a Sunday.
+
++
+For example,
+
+** The DATEDIFF difference is 1 between 2017-11-19 and 2017-11-26 if the specified _datepart_ is WEEK.
+
+** The DATEDIFF difference is 0 between 2017-11-19 and 2017-11-25 if the specified _datepart_ is WEEK.
+
+[[the_first_week_of_year]]
+==== The first week of year
+
+This value cannot be specified, the default value is the week in which Jan 1 occurs.  
+
+[[negative_or_zero]]
+==== Negative or zero
+
+If _enddate_ precedes _startdate_, the return value is negative or zero.
+
+[[error]]
+==== Error
+
+* The DATEDIFF function generates an error if the result is out of range for integer values. 
+
+** For seconds, the maximum number is equivalent to approximately 68 years. 
+
+** For weeks, if a difference in weeks is requested, one of the two dates cannot precede 0001-01-07.
 
 <<<
 [[examples_of_datediff]]
 === Examples of DATEDIFF
 
-* This function returns the value of 0 because no one-second boundaries
-are crossed.
+[[date_difference_in_second]]
+==== Date Difference in SECOND
+
+* This function returns the value of 0 because no one-second boundary
+is crossed.
 +
 ```
-DATEDIFF( SECOND
-        , TIMESTAMP '2006-09-12 11:59:58.999998'
-        , TIMESTAMP '2006-09-12 11:59:58.999999'
-        )
+SELECT DATEDIFF( SECOND
+                 , TIMESTAMP '2006-09-12 11:59:58.999998'
+                 , TIMESTAMP '2006-09-12 11:59:58.999999'
+               )
+FROM DUAL;
 ```
 
 * This function returns the value 1 because a one-second boundary is
 crossed even though the two timestamps differ by only one microsecond.
 +
 ```
-DATEDIFF( SECOND
-        , TIMESTAMP '2006-09-12 11:59:58.999999'
-        , TIMESTAMP '2006-09-12 11:59:59.000000'
-        )
+SELECT DATEDIFF( SECOND
+                 , TIMESTAMP '2006-09-12 11:59:58.999999'
+                 , TIMESTAMP '2006-09-12 11:59:59.000000'
+               )
+FROM DUAL;
 ```
 
-* This function returns the value of 0.
+[[date_difference_in_minute]]
+==== Date Difference in MINUTE
+
+* This function returns the value of 2 because a two-minute boundary is crossed.
 +
 ```
-DATEDIFF( YEAR
-        , TIMESTAMP '2006-12-31 23:59:59.999998'
-        , TIMESTAMP '2006-12-31 23:59:59.999999'
-        )
+SELECT DATEDIFF( MINUTE
+                 , TIMESTAMP '2011-12-30 08:54:55'
+                 , TIMESTAMP '2011-12-30 08:56:01'
+               ) 
+FROM DUAL;
+```
+
+[[date_difference_in_day]]
+==== Date Difference in DAY
+
+* This function returns the value of -29.
++
+```
+SELECT DATEDIFF( DAY
+                 , DATE '2008-03-01'
+                 , DATE '2008-02-01'
+               ) 
+FROM DUAL;
+```
+
+* This statement calculates how long do buyers have to wait. 
```

Comment:
Grammar. Suggest "This statement calculates how long buyers have to wait" (omit the word "do")

Reply:
Thanks Dave, your comments have been incorporated. :)